lib/board.ex

```diff
@@ -66,28 +49,96 @@ defmodule Board do
     end
   end
 
+  def board_full?(board) do
+    board
+    |> Enum.reject(&mark?/1)
+    |> Enum.empty?
+  end
+
+  def dimension(board) do
+    length(board)
+    |> :math.sqrt
+    |> round
+  end
+
+  def current_lines(board) do
+    [rows(board), columns(board), diagonals(board)]
+    |> List.flatten
+    |> Enum.chunk(dimension(board))
+  end
+
+  def rows(board), do: Enum.chunk(board, dimension(board))
+
+  defp columns(board) do
+    board
+    |> rows
+    |> transpose
+  end
+
+  defp diagonals(board) do
+    [
+      backward_diagonal([], board, dimension(board), dimension(board) - 1),
```

Comment:
Thanks, I've changed it to reverse the rows, so that I only need one function to create a diagonal. I'll look at the algorithm again if I have time left in the iteration.